manifests/plugin/apache/instance.pp

```diff
@@ -1,12 +1,12 @@
 # https://collectd.org/wiki/index.php/Plugin:Apache
 define collectd::plugin::apache::instance (
-  $url,
-  $ensure = present,
-  $user = undef,
-  $password = undef,
-  $verifypeer = undef,
-  $verifyhost = undef,
-  $cacert = undef,
+  Stdlib::Httpurl $url,
+  String $ensure                         = present,
+  Optional[String] $user                 = undef,
+  Optional[String] $password             = undef,
```

Comment:
I tried to use Sensitive, but couldn't get it working with `$password` being optional

Reply:
small steps, still better than no datatypes \o/